src/Compilers/Core/Portable/PEWriter/PeWriter.cs

```diff
@@ -115,6 +88,8 @@ private bool WritePeToStream(MetadataWriter mdWriter, Func<Stream> getPeStream,
             var managedResourceBuffer = new MemoryStream(1024);
             var managedResourceWriter = new BinaryWriter(managedResourceBuffer);
 
+            nativePdbWriterOpt?.SetMetadataEmitter(mdWriter);
```

Comment:
Unfortunately it uses it to get names for methods and types :( It's dumb, but it is what it is.